### PR TITLE
Update types & Implement GSP-654

### PR DESCRIPTION
**Pull request by JinnyYi**

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func (s *Storage) createAppend(ctx context.Context, path string, opt pairStorage
 	o.ID = rp
 	o.Path = path
 	o.SetAppendOffset(0)
+	o.SetContentLength(0)
```
Comment (Xuanwo) [inline storage.go:70]: Maybe we don't need to check `content-length` here? It's possible that appendable object doesn't have content-length, for example, dropbox?

Comment (JinnyYi) [reply storage.go:70]: > It's possible that appendable object doesn't have content-length, for example, dropbox?

Yes, we can't retrive the object and get the content-length by path until `CommitAppend`.

Can we consider that a successful `CreateAppend` operation should create a new appendable object with position 0 and size 0?

Comment (Xuanwo) [reply storage.go:70]: I think only offset is required. Maybe we need to fix it in our integration tests?

Comment (JinnyYi) [reply storage.go:70]: Got it.

**Comment by Xuanwo** [comment]
Maybe we can wait for discussion from https://forum.beyondstorage.io/t/how-to-maintain-go-service-more-easily/137

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**